airbyte-ci/connectors/metadata_service/lib/metadata_service/stale_metadata_report.py

```diff
@@ -0,0 +1,140 @@
+    storage_client = get_gcs_storage_client()
+    bucket = storage_client.bucket(bucket_name)
+
+    blobs: list[storage.Blob] = bucket.list_blobs(match_glob=f"{METADATA_FOLDER}/**/latest/{METADATA_FILE_NAME}")
+
+    latest_metadata_entries_on_gcs = {}
+    for blob in blobs:
+        metadata_dict = yaml.safe_load(blob.download_as_string().decode("utf-8"))
+        latest_metadata_entries_on_gcs[metadata_dict["dockerRepository"]] = metadata_dict["dockerImageTag"]
+
+    return latest_metadata_entries_on_gcs
+
+
+def _generate_stale_metadata_report(
+    latest_metadata_versions_on_github: Mapping[str, Any], latest_metadata_entries_on_gcs: Mapping[str, Any]
+) -> pd.DataFrame:
+    stale_connectors = []
+    for docker_repository, github_docker_image_tag in latest_metadata_versions_on_github.items():
+        gcs_docker_image_tag = latest_metadata_entries_on_gcs.get(docker_repository)
+        if gcs_docker_image_tag != github_docker_image_tag:
+            stale_connectors.append(
+                {"connector": docker_repository, "master_version": github_docker_image_tag, "gcs_version": gcs_docker_image_tag}
+            )
+
+    return pd.DataFrame(stale_connectors)
+
+
+def generate_and_publish_stale_metadata_report(bucket_name: str) -> tuple[bool, str]:
+    """
+    Generate a stale metadata report and publish it to a Slack channel.
+
+    Args:
+        bucket_name (str): The name of the GCS bucket to check for stale metadata.
+
+    Returns:
+        bool: True if any stale metadata was detected, False otherwise.
+    """
+
+    latest_metadata_versions_on_github = _get_latest_metadata_versions_on_github()
+    latest_metadata_entries_on_gcs = _get_latest_metadata_entries_on_gcs(bucket_name)
+    stale_metadata_report = _generate_stale_metadata_report(latest_metadata_versions_on_github, latest_metadata_entries_on_gcs)
+
+    stale_report_channel = os.getenv("STALE_REPORT_CHANNEL")
+    publish_update_channel = os.getenv("PUBLISH_UPDATE_CHANNEL")
+    any_stale = len(stale_metadata_report) > 0
+    if any_stale and stale_report_channel:
+        stale_report_md = stale_metadata_report.to_markdown(index=False)
+        send_slack_message(stale_report_channel, f"🚨 Stale metadata detected! (cc. <!subteam^{EXTENSIBILITY_TEAM_SLACK_TEAM_ID}>)")
+        send_slack_message(stale_report_channel, stale_report_md, enable_code_block_wrapping=True)
+
+    if not any_stale and publish_update_channel:
+        message = textwrap.dedent(
+            f"""
+        Analyzed {len(latest_metadata_versions_on_github)} metadata files on our master branch and {len(latest_metadata_entries_on_gcs)} latest metadata files hosted in GCS.
+        All dockerImageTag value on master match the latest metadata files on GCS.
+        No stale metadata: GCS metadata are up to date with metadata hosted on GCS.
+        """
+        )
+        send_slack_message(publish_update_channel, message)
```

Comment:
is this just to match existing behavior? It feels weird to me to send a message saying "everything is fine", I generally prefer to only send messages when there's something wrong